### PR TITLE
application credentials email/password

### LINKED ISSUE #392: Support App Credentials email\password
realm_app_credentials_new_email_password

### PR DESCRIPTION
**Pull request by desistefanova**
Implemented application credentials email/password. Fixes #392 
` final credentials = Credentials.emailPassword("test@email.com", "000000");`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ x.x.x Release notes (yyyy-MM-dd)
 
 ### Enhancements
 * Support anonymouse application credentials ([#443](https://github.com/realm/realm-dart/pull/443/))
```
Comment (blagoev) [inline CHANGELOG.md:7]:
```suggestion
* Support anonymous application credentials ([#443](https://github.com/realm/realm-dart/pull/443/))
```

**Review by blagoev — CHANGES_REQUESTED** [review]

File: lib/src/native/realm_core.dart [hunk 2]
```diff
@@ -762,6 +770,18 @@ extension _RealmLibraryEx on RealmLibrary {
   }
 }
 
+Pointer<realm_string_t> _toRealmString(String value, Allocator allocator) {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply lib/src/native/realm_core.dart:773]: OK.

**Review by blagoev — APPROVED** [review]

File: lib/src/app_credentials.dart [hunk 3]
```diff
@@ -41,6 +41,14 @@ class Credentials {
   Credentials.anonymous()
       : _handle = realmCore.createAppCredentialsAnonymous(),
         provider = AuthProvider.anonymous;
+
+  /// Returns a [Credentials] object that can be used to authenticate a user with their email and password.
+  /// A user can login with email and password only after they have registered their account and verified their
+  /// email.
+  /// [Email/Password Authentication Docs](https://docs.mongodb.com/realm/authentication/email-password)
+  Credentials.emailPassword(String email, String password)
+      : _handle = realmCore.createAppCredentialsEmailPassword(email, password),
```
Comment (nielsenko) [inline lib/src/app_credentials.dart:50]: This ties into @nirinchev's proposal, that we should not create the handle until actually needed. If we do agree on that then this needs to change somewhat.. 

Fx. we could make a subclass `EmailAndPassworsCredentials` that has two final fields (email & password) and have `_handle` be late with same initializer as here.

But again.. depends on what stance we take on @nirinchev 's PR..

File: lib/src/native/realm_core.dart [hunk 4]
```diff
@@ -743,6 +751,14 @@ extension _StringEx on String {
     nativeString.last = 0; // zero terminate
     return result.cast();
   }
+
+  Pointer<realm_string_t> toRealmString(Allocator allocator) {
```
Comment (nielsenko) [inline lib/src/native/realm_core.dart:755]: I feel this belongs somewhere else.. not really relates to `LastError`

File: lib/src/app_credentials.dart (hunk 3 above)
Comment (nielsenko) [reply lib/src/app_credentials.dart:50]: To untangle the PR jam, I think we should merge as is. We can always revisit

**Review by nielsenko — APPROVED** [review]
May or may not refactor later depending on: #455

**Review by nirinchev — APPROVED** [review]
This looks good to me, but why is the Core API accepting different types for the email and password? Seems kinda weird.

File: lib/src/app_credentials.dart (hunk 3 above)
Comment (desistefanova) [reply lib/src/app_credentials.dart:50]: Yes, we can change it later.

File: lib/src/native/realm_core.dart (hunk 4 above)
Comment (desistefanova) [reply lib/src/native/realm_core.dart:755]: It is in `extension _StringEx on String {`

**Pull request merged**